x/crosschain/keeper/msg_server_migrate_tss_funds.go

```diff
@@ -27,8 +27,9 @@ func (k msgServer) MigrateTssFunds(goCtx context.Context, msg *types.MsgMigrateT
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	// check if authorized
-	if !k.GetAuthorityKeeper().IsAuthorized(ctx, msg.Creator, authoritytypes.PolicyType_groupAdmin) {
-		return nil, errorsmod.Wrap(authoritytypes.ErrUnauthorized, "Update can only be executed by the correct policy account")
+	ok, err := k.GetAuthorityKeeper().IsAuthorized(ctx, msg)
+	if !ok || err != nil {
+		return nil, errorsmod.Wrap(authoritytypes.ErrUnauthorized, err.Error())
```

Comment:
This would panic if ok == false, err == nil. Has it been tested?

Reply:
How would that happen? The function always returns an error message if the `IsAuthorized` Fails. The error message is used to add some more context to why the authorization failed.

Reply:
It's an antipattern to return other variables with the error, if the function returns an error it should be assumed that the other returned values should not be used, if custom values are returned, it should be assumed err is nil.

We should change the error into a string `nonAuthorizedReason` or remove the boolean and just return an error when not auithorized

Reply:
d3741a6e5e58747e2fa9cd6f39ece7d6ca0f0a37